docs/specifications/tests/MethodsV2.md

```diff
@@ -0,0 +1,1026 @@
+# Methods common to Test Case Specifications (version 2)
+
+## Table of contents
+
+* [Objective](#objective)
+* [Scope](#scope)
+* [Internal Methods]
+* [Method: Get parent NS IP addresses][Get-Parent-NS-IP]
+* [Method: Get delegation NS names and IP addresses][Get-Del-NS-Names-and-IPs]
+* [Method: Get delegation NS names][Get-Del-NS-Names]
+* [Method: Get delegation NS IP addresses][Get-Del-NS-IPs]
+* [Method: Get zone NS names][Get-Zone-NS-Names]
+* [Method: Get zone NS names and IP addresses][Get-Zone-NS-Names-and-IPs]
+* [Method: Get zone NS IP addresses][Get-Zone-NS-IPs]
+* [Method: Get delegation (Internal)][Get-Delegation]
+* [Method: Get in-bailiwick address records in zone (Internal)][Get-IB-Addr-in-Zone]
+* [Method: Get out-of-bailiwick ip addresses (Internal)][Get-OOB-IPs]
+* [Method: Get data for undelegated test (Internal)][Get-Undel-Data]
+* [Method inter-dependencies](#method-inter-dependencies)
+* [Terminology](#terminology)
+
+
+## Objective
+
+The Methods are used in, and referred from, the Test Case specifications as
+shortcuts for steps shared between Test Cases. A Test Case that makes use of any
+of the Methods defined in this document must refer directly to the
+specific Method or Methods.
+
+[To top]
+
+
+## Scope
+
+This document holds version 2 of the set of Methods. Version 1 is defined in
+[Methods]. Methods from version 1 will be replaced by Methods from version 2 in
+all Test Case specifications.
+
+Before the transition all Test Cases specifications use version 1 ([Methods]).
+During the transition it will be stated in each specification if the
+Test Case uses Methods from version 1 ([Methods]) or Methods from version 2 (this
+document). When the transition is completed, the version 1 document will be
+removed.
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in
+[DNS Query and Response Defaults]. The handling of the DNS responses on the
+DNS queries follow, unless otherwise specified below, what is specified for
+[DNS Response] in the same specification.
+
+In these Methods any DS record data in *Undelegated Data* is disregarded. If
+with *Child Zone* DS record data is provided, but no name server data, then the
+will here be treated as "normal test", not "undelegated test".
+
+[To top]
+
+
+## Internal methods
```

Comment:
This section is not relevant to people who only consume this document. It is only relevant to people defining methods and test cases.
Maybe rename this section to "On defining methods and test cases" or something?

Reply:
It would be less clear for whom?

Reply:
Some methods are tagged as internal. Here is a section that explains that.